mitodl/mit-open
# ability to set search mode

### What are the relevant tickets?
closes https://github.com/mitodl/hq/issues/5181

### Description (What does it do?)
This pr adds settings to the admin panel to allow users to select the search mode for queries. The user can select the search type from https://opensearch.org/docs/latest/query-dsl/full-text/multi-match/. Additionally if the user selects a "phrase" as the query type, the user can set the "slop" which is the number of words that are allowed between the terms in a multi word query. 

Currently this setting only affects text queries that are not enclosed in quotes.

### Screenshots (if appropriate):
<img width="1728" alt="Screenshot 2024-08-23 at 4 02 38 PM" src="https://github.com/user-attachments/assets/78d3c37e-7386-4b8a-aa39-3dbdf1fce077">
<img width="1726" alt="Screenshot 2024-08-23 at 4 02 53 PM" src="https://github.com/user-attachments/assets/4eb2337b-1671-46de-ab0c-4c0a1722c987">
<img width="327" alt="Screenshot 2024-08-23 at 4 03 36 PM" src="https://github.com/user-attachments/assets/ef8e0072-ccc0-4b5b-a737-b3483d3d16df">

### How can this be tested?
Go to open.odl.local:8062/search/ as an admin user
Verify that you can select search_mode from the dropdown
Verify that if you select search_mode=phrase you can set slop with the slider
Verify that the slop selector is not shown and slop is removed from the url if you select a different search mode
Verify that if there is a text query, your selections affect the results you see.

### Additional Context
Most likely we will want to change the default sort to search_mode=phrase and slop something like 5-10. This pr will allow us to play with the results on prod before making that change
